### ISSUE TITLE
Message delivery problem

### ISSUE THREAD

**Issue by kriztan** [siacs/Conversations#863]
I'm using latest prosody trunk and activated inter alia MAM and Stream Management. All my contacts are using Conversations 0.10. Now I experienced that there are sometimes "delivery failed" errors despite of activated MAM and Stream Management. I checked the receivers, they received my "failed" messages. So the "delivery failed" is a faulty information for the sender. 
The delivery status should distinguish between received via MAM and "normal" way. Is Stream Management even still required with activated MAM?

**Comment by iNPUTmice** [comment]
As explained in the [README](https://github.com/siacs/Conversations#i-get-delivery-failed-on-my-messages) delivery failed is something the server reports.
If a stream management session dies prosody still returns undelivered messages to sender because the sm module doesn't know about the mam module. The sender has no ability at all to know whether the recipient has MAM or not.
TLDR: this is something prosody should fix by allowing the user to disable that behavior. (like ejabberd does for example)

>  Is Stream Management even still required with activated MAM?

It is a lot more work to establish a complete session (fetching roster, fetching blocklist, joining rooms, catching up with MAM) than to resume a session. This is not something we want to do on every connectivity change. So yes. SM is still something we need.
